pytensor/graph/rewriting/utils.py

```diff
@@ -83,8 +84,10 @@ def is_same_graph_with_merge(var1, var2, givens=None):
 
     if givens is None:
         givens = {}
+    givens = dict(givens)
+
     # Copy variables since the MergeOptimizer will modify them.
-    copied = copy.deepcopy([var1, var2, givens])
+    copied = copy.deepcopy((var1, var2, givens))
```

Comment:
Nit: Perhaps inline?

```suggestion
    copied = copy.deepcopy((var1, var2, dict(givens)))
```
And get rid of `givens = dict(givens)` above
```suggestion
    copied = copy.deepcopy((var1, var2, givens))
```

Reply:
> And get rid of `givens = dict(givens)` above and the unpacking below

`givens` can be a list of 2-ples, a tuple of 2-ples, or a dictionary. It looks difficult to treat these three types equally here without converting them into one of them.

Actually, these two funtions `is_same_graph` and `is_same_graph_with_merge` are only used in the tests. Perhaps I should even move them there? Or do they have an interest for pytensor users?

Reply:
Those functions sound useful for users, assuming they have a reasonable implementation

Reply:
What is the problem with my change suggestion? You still convert it to a dict just not save the intermediate result. And it was being over shadowed by the output of the deepcopy anyway

Reply:
Ah yes indeed, I missed one part of your suggestion

Reply:
In the meantime I found
``` python
givens = {} if givens is None else dict(givens)
```
which is maybe alright? It also echoes the same line in the other function

Reply:
sounds good